### ISSUE TITLE
Rspack alias not working as expected

### ISSUE THREAD

**Issue by dgmachado** [web-infra-dev/rspack#2914]
### System Info

System:
OS: macOS 13.1
CPU: (10) arm64 Apple M1 Pro
Memory 32GB
Shell: /bin/zsh

Binaries: 
node: 16.13.2
npm: 9.5.0
yarn: 1.22.19

Browsers:
Chrome: 111.0.5563.146

npmPackages:
@nrwl/rspack: 15.9.5
@rspack/core: ~0.1.0
@rspack/dev-server: ~0.1.0
@rspack/plugin-minify: ~0.1.0

### Details

The Rspack alias isn't working as expected.  As you can see below only scenario 1 works.

```javascript
resolve: {
    alias: {
      // # Scenario 1
      //  It works
      './environments/environment': '/..{absolutePathToTheApp}/nestapp/src/environments/environment.prod.ts',
      //
      // # Scenario 2 
      // it should work also but it doesn't 
      './environments/environment': './environments/environment.prod.ts',
      //
      // # Scenario 3 (relative path)
      // it should work also but it doesn't 
      './environments/environment.ts': './environments/environment.prod.ts',
    }
  }
```

### Reproduce Steps

I've created a [sample app](https://github.com/dgmachado/nestapp-example/tree/build) you can use the branch "build" and to test you can execute the below commands:

```
npm run build-rspack
npm run serve-rspack
```

What is expected:
```
Environment.production:true
```
What is appearing:
```
Environment.production:false
```

**Comment by hyf0** [comment]
I would suggest using https://github.com/web-infra-dev/rspack-repro to confirm whether Webpack works as expected.

**Comment by dgmachado** [comment]
@hyf0 I just created two alias scenarios not working in Rspack as expected. I just invited you as a contributor on the [repo](https://github.com/dgmachado/rspack-webpack-repro/tree/bug/rspack-alias). In summary, the below scenarios are working in Webpack and not in Rspack.

```
const questionAbsolutePath = path.resolve(__dirname, "./src/question.js")

...
  resolve: {
    alias: {
     "./answer": path.resolve(__dirname, "./src/answer.prod.js?raw"),
     questionAbsolutePath: path.resolve(__dirname, "./src/question.prod.js?raw"),
    },
  },
```

**Comment by hyf0** [comment]
Thank you for taking time to create the repo. It's a bug of Rspack and we plan to fix it.

**Comment by dgmachado** [comment]
> @dgmachado Excuse me for the late reply, can you also add me as a contributor? I'll try and unblock this for you during the next week.

@Boshen sure. I just invited you.

**Comment by Boshen** [comment]
@dgmachado Your repros are some what confusing for me, you are mixing a few scenarios in different contexts.

For your real app, which one exact is broken?

If it's scenario 3 from the first nest repo, I can't replicate it with Webpack. This is done by changing everything to `.ts`:

```
ERROR in ./src/index.ts 2:0-38
Module not found: Error: Can't resolve './question' in '/Users/bytedance/github/rspack-webpack-repro/src'
resolve './question' in '/Users/bytedance/github/rspack-webpack-repro/src'
  using description file: /Users/bytedance/github/rspack-webpack-repro/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: /Users/bytedance/github/rspack-webpack-repro/package.json (relative path: ./src/question)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /Users/bytedance/github/rspack-webpack-repro/src/question doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /Users/bytedance/github/rspack-webpack-repro/src/question.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        /Users/bytedance/github/rspack-webpack-repro/src/question.json doesn't exist
      .wasm
        Field 'browser' doesn't contain a valid alias configuration
        /Users/bytedance/github/rspack-webpack-repro/src/question.wasm doesn't exist
      as directory
        /Users/bytedance/github/rspack-webpack-repro/src/question doesn't exist
```

----

If it's this example:

```
   alias: {
     questionAbsolutePath: path.resolve(__dirname, "./src/question.prod.js?raw"),
    },
```

It's not going to work because it's not evaluated, did you mean 

```
   alias: {
     [questionAbsolutePath]: path.resolve(__dirname, "./src/question.prod.js?raw"),
    },
```

But this is the not same as scenario as the previous one?

I'm getting super confused here.

**Comment by dgmachado** [comment]
@Boshen thanks for the feedback. I just pushed a new commit into the [repo](https://github.com/dgmachado/rspack-webpack-repro/tree/bug/rspack-alias) repo and branch bug/rspack-alias.

In my scenario, I need the below config working as webpack:

```
[questionAbsolutePath]: path.resolve(__dirname, "./src/question.prod.js"),
```

You can compare the results by executing the below commands:
```
npm run build:rspack
npm run build:webpack
```

**Comment by Boshen** [comment]
Thank you for the confirmation, linking to 
* https://github.com/web-infra-dev/nodejs_resolver/issues/185

**Comment by bvanjoi** [comment]
Thanks for the report. The issue was caused by a lack of re-doResolve process after normalizing the request. Although it is a bug and needs to be fixed, it can be avoided by using a relative alias key. Therefore, it has been given a low priority.

**Comment by dgmachado** [comment]
> Thanks for the report. The issue was caused by a lack of re-doResolve process after normalizing the request. Although it is a bug and needs to be fixed, it can be avoided by using a relative alias key. Therefore, it has been given a low priority.

@bvanjoi thank you for the heads up.

**Comment by hardfist** [comment]
@dgmachado this bug is fixed in new resolver implmentation, you can test by  enable new resolver

```js
// rspack.config.js
module.exports = {
  experiments: {
   rspackFuture: {

  newResolver: true
}
  }
}
```
and you can see a example here https://github.com/hardfist/rspack-absolute-alias/blob/fix-alias/config.mjs#L39
